cabal-install/bootstrap.sh

```diff
@@ -52,24 +55,32 @@ SCOPE_OF_INSTALLATION="${SCOPE_OF_INSTALLATION:---user}"
 DEFAULT_PREFIX="${HOME}/.cabal"
 
 # Try to respect $TMPDIR.
-[ -"$TMPDIR"- = -""- ] &&
-  export TMPDIR=/tmp/cabal-$(echo $(od -XN4 -An /dev/random)) && mkdir $TMPDIR
+if [ -"$TMPDIR"- = -""- ]; then
+    TMPDIR=/tmp/cabal-$(od -XN4 -An /dev/random)
+    mkdir "$TMPDIR"
+    export TMPDIR
+fi
```

Comment:
@23Skidoo, a couple of questions regarding TMPDIR:

1. Where is this used? Is the env var reused by cabal-install?
2. It goes against convention to name it TMPDIR because you set TMPDIR to deviate from /tmp, so referencing a directory inside /tmp via TMPDIR isn't right.

How about this instead: `mktemp -d -p /tmp -t cabal-XXXXXXXX`. This is portable to OpenBSD, FreeBSD, Linux and Solaris. The parent dir falls back to /tmp (if given via -p or assumed to be default if not) if TMPDIR is not set. The result will look like `/tmp/cabal-XYXYXYXY`.

Because I didn't figure out where TMPDIR is used in cabal and to see if this is the cause of the 7.10.3 build failure, I'll add a test commit.